great_expectations/experimental/datasources/config.py

```diff
@@ -61,3 +79,34 @@ def _load_datasource_subtype(cls, v: Dict[str, dict]):
         if v and not loaded_datasources:
             raise ValueError(f"Of {len(v)} entries, no 'datasources' could be loaded")
         return loaded_datasources
+
+    @classmethod
+    def parse_yaml(
+        cls: Type[GxConfig], f: Union[pathlib.Path, str], _allow_empty: bool = False
+    ) -> GxConfig:
+        """
+        Overriding base method to allow an empty/missing `xdatasources` field.
+        Other validation errors will still result in an error.
+
+        TODO (kilo59) 122822: remove this as soon as it's no longer needed. Such as when
+        we use a new `config_version` instead of `xdatasources` key.
+        """
+        if _allow_empty:
+            try:
+                super().parse_yaml(f)
+            except ValidationError as validation_err:
+                errors_list: List[PydanticErrorDict] = validation_err.errors()
+                LOGGER.info(
+                    f"{cls.__name__}.parse_yaml() failed with errors - {errors_list}"
+                )
+                if errors_list == _MISSING_XDATASOURCES_ERRORS:
+                    LOGGER.info(
+                        f"{cls.__name__}.parse_yaml() returning empty `xdatasources`"
+                    )
+                    return cls(xdatasources={})
+                else:
+                    LOGGER.warning(
+                        "`_allow_empty` does not prevent unrelated validation errors"
+                    )
+                    raise
+        return super().parse_yaml(f)
```

Comment:
I really don't like this special casing...
But my other ideas would have been much more invasive.
And hopefully very soon we can get rid of this + the `xdatasources` field and use the `config_version` to control which style of config to parse.